tpp-app/src/services/CalculationService.js

```diff
@@ -0,0 +1,167 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import {FLOW_LEVEL, KEYS} from "./utils/constants";
+import subDays from 'date-fns/subDays';
+import addDays from 'date-fns/addDays';
+import {getCalendarByYear, getDaysDiffInclusive, getSymptomsFromCalendar} from "./utils/helpers";
+import CycleService from "./cycle/CycleService";
+import { GETStoredYears } from "./utils/helpers";
+
+
+/**
+ * Returns Array of the Symptoms of the last three days
+ * @return {Array} of size 3
+ */
+const GETLastThreeDaysSymptoms = async () => {
+    const today = new Date();
+    const yesterday = subDays(today, 1);
+    const twoDaysEarlier = subDays(today, 2);
+
+    var calendar = await getCalendarByYear(today.getFullYear());
+    const todaySymptoms = await getSymptomsFromCalendar(
+        calendar,
+        today.getDay(),
+        today.getMonth() + 1,
+        today.getFullYear()
+    );
+
+    if (today.getFullYear() !== yesterday.getFullYear()) {
+        calendar = await getCalendarByYear(yesterday.getFullYear());
```

Comment:
you shouldn't need to check whether `today`, `yesterday` and `twoDaysEarlier` are in the same year. `getCalendarByYear` is a dict with the calendars for the current year, prev year & future year (not included in dict if they aren't in AsyncStorage).

Since `getSymptomsFromCalendar` uses this calendar, it'll still return the symptoms for the correct date as long as the date is in {prev year, current year, next year}

Reply:
OH RIGHT i remember asking u about it and then i didn't change 😅 ok updated